ipam/allocator.go

```diff
@@ -46,9 +77,11 @@ type Allocator struct {
 // NewAllocator returns an instance of libnetwork ipam
 func NewAllocator(ds datastore.DataStore) (*Allocator, error) {
 	a := &Allocator{}
-	a.subnets = make(map[subnetKey]*SubnetInfo)
+	a.subnets = make(map[subnetKey]*PoolData)
 	a.addresses = make(map[subnetKey]*bitseq.Handle)
-	a.internalHostSize = defaultInternalHostSize
+	a.predefined = make(map[string][]*net.IPNet, 2)
+	a.predefined[localAddressSpace] = localPredefinedPools
+	a.predefined[globalAddressSpace] = globalPredefinedPools
 	a.store = ds
 	a.App = "ipam"
 	a.ID = dsConfigKey
```

Comment:
Do we really need this ?
The address space is already part of the individual configuration object's key when inserted in the DS.
Because of that the subnetKey are not colliding, this allows to keep the config for all address spaces in single level map `subnets map[subnetKey]*PoolData`.

Also, `ID` belongs to `Allocator`. I am not going to have an allocator per address space.